### PR TITLE
Fix fasm CLBLM and SING tiles

### PR DESCRIPTION
**Pull request by mtdudek**
This PR fixes routing issues inside SLICEM sites, SRL muxes and some other signals related to distributed memory weren't properly handled. Also IO SING tiles had issue. These tiles only have 1 site, and its bits depend on tile position in the fabric grid.
For tiles that their Y mod 50 is zero, site uses bits for site Y0 but if Y mod 50 is 49 then site is configured using Y1 bits.
https://github.com/SymbiFlow/prjxray/blob/cb5f2a30633bf0d3ac91a9b5802790ac1da86736/fuzzers/005-tilegrid/generate_full.py#L357-L365

### CODE REVIEW

File: fpga_interchange/rapidyaml_support.py [hunk 1]
```diff
@@ -91,6 +91,8 @@ def make_reference(self, ref_id):
             return YamlReference(ref)
 
     def get_list_id(self, field, index):
+        if field not in self.field_ids.keys():
+            return 0
```
Comment (mkurc-ant) [inline fpga_interchange/rapidyaml_support.py:95]: Can you elaborate on why this change is required ?

**Comment by mkurc-ant** [comment]
@mtdudek Please add a short description of what are the things that this PR fixes.

File: fpga_interchange/rapidyaml_support.py (hunk 1 above)
Comment (mtdudek) [reply fpga_interchange/rapidyaml_support.py:95]: This change fixes invalid key if capnp file didn't contain some fields. I think that this broke for device schema.

**Review by mkurc-ant — APPROVED** [review]
LGTM
